libpod/healthcheck.go

```diff
@@ -145,8 +168,23 @@ func newHealthCheckLog(start, end time.Time, exitCode int, log string) inspect.H
 	}
 }
 
+// updatedHealthCheckStatus updates the health status of the container
+// in the healthcheck log
+func (c *Container) updateHealthStatus(status string) error {
+	healthCheck, err := c.GetHealthCheckLog()
+	if err != nil {
+		return err
+	}
+	healthCheck.Status = status
+	newResults, err := json.Marshal(healthCheck)
+	if err != nil {
+		return errors.Wrapf(err, "unable to marshall healthchecks for writing status")
+	}
+	return ioutil.WriteFile(c.healthCheckLogPath(), newResults, 0700)
+}
+
 // UpdateHealthCheckLog parses the health check results and writes the log
-func (c *Container) updateHealthCheckLog(hcl inspect.HealthCheckLog) error {
+func (c *Container) updateHealthCheckLog(hcl inspect.HealthCheckLog, inStartPeriod bool) error {
```

Comment:
Why not make inStartPeriod part of the Container struct?

Reply:
because startPeriod is related to the start time of the container check (gotten with time.Now), i pass it because the check occurred technically in the past relative to updateHealthCheckLog ... if you don't care about the amount of time elapsed between when the healthcheck got called and when we write the log, it surely can be done.

Reply:
If that explanation was in the code, I would not have had to ask.  :^)